pytest_tests/testsuites/object/test_object_api.py

```diff
@@ -337,6 +341,69 @@ def test_object_search_should_return_tombstone_items(
                     object_type == "TOMBSTONE"
                 ), f"Object wasn't deleted properly. Found object {tombstone_oid} with type {object_type}"
 
+    @allure.title("Validate objects search by common prefix")
+    def test_search_object_api_common_prefix(self, default_wallet: str, simple_object_size: int):
+        FILEPATH_ATTR_NAME = "FilePath"
+        cli = NeofsCli(self.shell, NEOFS_CLI_EXEC, WALLET_CONFIG)
+        wallet = default_wallet
+        cid = create_container(wallet, shell=self.shell, endpoint=self.cluster.default_rpc_endpoint)
+
+        objects = {}
+        for _ in range(5):
+            file_path = generate_file(simple_object_size)
+
+            with allure.step("Put objects"):
+                objects[file_path] = put_object_to_random_node(
+                    wallet=wallet,
+                    path=file_path,
+                    cid=cid,
+                    shell=self.shell,
+                    cluster=self.cluster,
+                    attributes={FILEPATH_ATTR_NAME: file_path}
+                )
+        all_oids = sorted(objects.values())
+
+        for common_prefix, expected_oids in (
+            ('/', all_oids),
+            (os.path.join(os.getcwd(), ASSETS_DIR), all_oids),
+            (os.path.join(os.getcwd(), ASSETS_DIR, TEST_FILES_DIR), all_oids),
+            (file_path, objects[file_path])
+        ):
+            with allure.step(f"Search objects by path: {common_prefix}"):
+                result = cli.object.search(
+                    rpc_endpoint=self.cluster.default_rpc_endpoint,
+                    wallet=wallet,
+                    cid=cid,
+                    filters=[f"{FILEPATH_ATTR_NAME} COMMON_PREFIX {common_prefix}"],
+                    root=True,
+                )
+
+                found_oids = re.findall(r"(\w{43,44})", result.stdout)
+                if len(found_oids) > 1:
+                    assert expected_oids == sorted(found_oids)
+                elif len(found_oids):
+                    assert expected_oids == found_oids[0]
+                else:
+                    raise AssertionError(f"No objects found by common prefix: {common_prefix}")
+        
+        for common_prefix in (
```

Comment:
Maybe this would be better organized as an enumeration?

Reply:
no, it wouldn't =) it is just test cases inside this particular test, it can be used inside a parametrization decorator, but I decided to move it here to speed up test a little bit